File: medbrains/apps/android/ui/src/main/kotlin/com/medbrains/ui/Brand.kt
package com.medbrains.ui

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.CornerRadius
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Rect
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Path
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.StrokeJoin
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/** The MedBrains mark, drawn — the same five shapes as the web's medbrains-mark.svg. */
@Composable
fun BrandMark(size: Dp = 40.dp, modifier: Modifier = Modifier) {
    val blue = Carbon.blue[5]
    val green = Carbon.mint[3]
    Canvas(modifier.size(size)) {
        val s = this.size.width / 44f
        drawRoundRect(blue, cornerRadius = CornerRadius(3 * s))
        drawArc(Color.White.copy(alpha = 0.3f), 180f, 180f, false, topLeft = Offset(9 * s, 16 * s), size = Size(26 * s, 26 * s), style = Stroke(1.25f * s, cap = StrokeCap.Round))
        val trace = Path().apply {
            moveTo(7 * s, 29 * s)
            listOf(14f to 29f, 16.5f to 24f, 18.5f to 33f, 21f to 16f, 23.5f to 29f, 29f to 29f).forEach { (x, y) -> lineTo(x * s, y * s) }
        }
        drawPath(trace, Color.White, style = Stroke(2 * s, cap = StrokeCap.Round, join = StrokeJoin.Round))
        drawLine(green, Offset(29 * s, 29 * s), Offset(35 * s, 29 * s), strokeWidth = 2 * s, cap = StrokeCap.Round)
        drawCircle(green, radius = 2 * s, center = Offset(29 * s, 29 * s))
    }
}

/** Mark plus the name, in the brand face. The login screens open with it. */
@Composable
fun BrandWordmark(height: Dp = 32.dp, modifier: Modifier = Modifier) {
    Row(modifier.semantics(mergeDescendants = true) { contentDescription = "MedBrains" }, verticalAlignment = Alignment.CenterVertically) {
        BrandMark(height)
        Spacer(Modifier.width(height * 0.3f))
        Text("MedBrains", fontFamily = PlexSans, fontWeight = FontWeight.SemiBold, fontSize = (height.value * 0.62f).sp, letterSpacing = (-0.5).sp, color = MaterialTheme.colorScheme.onBackground)
    }
}

File: medbrains/apps/ios/Packages/MedBrainsUI/Sources/MedBrainsUI/Brand.swift
import SwiftUI

/// The MedBrains mark, drawn — the same five shapes as
/// apps/web/public/logo/medbrains-mark.svg, so it is crisp at any size and
/// carries no bitmap. Blue 60 ground, the trace in white, the pulse in Green 40.
public struct BrandMark: View {
    let size: CGFloat
    public init(size: CGFloat = 40) { self.size = size }

    public var body: some View {
        Canvas { context, box in
            let s = box.width / 44
            var ground = Path(roundedRect: CGRect(x: 0, y: 0, width: box.width, height: box.height), cornerRadius: 3 * s)
            context.fill(ground, with: .color(Carbon.blue[5]))
            ground = Path()
            var arc = Path()
            arc.addArc(center: CGPoint(x: 22 * s, y: 29 * s), radius: 13 * s, startAngle: .degrees(180), endAngle: .degrees(0), clockwise: false)
            context.stroke(arc, with: .color(.white.opacity(0.3)), style: StrokeStyle(lineWidth: 1.25 * s, lineCap: .round))
            var trace = Path()
            trace.move(to: CGPoint(x: 7 * s, y: 29 * s))
            for p in [(14, 29), (16.5, 24), (18.5, 33), (21, 16), (23.5, 29), (29, 29)] { trace.addLine(to: CGPoint(x: CGFloat(p.0) * s, y: CGFloat(p.1) * s)) }
            context.stroke(trace, with: .color(.white), style: StrokeStyle(lineWidth: 2 * s, lineCap: .round, lineJoin: .round))
            var tail = Path()
            tail.move(to: CGPoint(x: 29 * s, y: 29 * s))
            tail.addLine(to: CGPoint(x: 35 * s, y: 29 * s))
            context.stroke(tail, with: .color(Carbon.mint[3]), style: StrokeStyle(lineWidth: 2 * s, lineCap: .round))
            context.fill(Path(ellipseIn: CGRect(x: 27 * s, y: 27 * s, width: 4 * s, height: 4 * s)), with: .color(Carbon.mint[3]))
        }
        .frame(width: size, height: size)
        .accessibilityHidden(true)
    }
}

/// Mark plus the name, in the brand face. The login screens open with it.
public struct BrandWordmark: View {
    let height: CGFloat
    public init(height: CGFloat = 32) { self.height = height }

    public var body: some View {
        HStack(spacing: height * 0.3) {
            BrandMark(size: height)
            Text("MedBrains")
                .font(.custom("IBMPlexSans-SemiBold", size: height * 0.62, relativeTo: .title2))
                .tracking(-0.5)
                .foregroundStyle(MedBrainsTheme.ink)
        }
        .accessibilityElement(children: .ignore)
        .accessibilityLabel("MedBrains")
    }
}
